keycloak/sms-provider/src/main/java/org/sunbird/keycloak/storage/spi/UserSearchService.java

```diff
@@ -30,7 +30,7 @@ public static List<User> getUserByKey(String key, String value) {
     Map<String, Object> request = new HashMap<>();
     request.put("key",key.toLowerCase());
     request.put("value", value);
-    request.put("fields", Arrays.asList("email","firstName","lastName","id","phone","userName","countryCode","status"));
+    request.put("fields", Arrays.asList("email","firstName","lastName","id","phone","userName","countryCode","status","rootorgid","roles"));
```

Comment:
In which API request these fields used?
Have you tested the API to return these values ?
roles -- are not returned mostly... do confirm that the used API return roles value...

Reply:
this is used for the auth-token generation (included orgID and userRoles)  which fetch details from lms service /private/user/v1/lookup  api,
the api is enhanced to fetch the roles from user roles table if the fields contain roles.
we have tested these api's.